### ISSUE TITLE
Compiler Warning in Xcode 8.3 (Swift 3.1)

### ISSUE THREAD

**Issue by chessboy** [SwiftyJSON/SwiftyJSON#826]
### What did you do?

Compiled project using SwiftyJSON library

### What did you expect to happen?

Expecting not to get a compiler warning

### What actually happened instead?

SwiftyJSON.swift:808:27: Expression implicitly coerced from 'String?' to Any

### Environment

List the software versions you're using:

 - SwiftyJSON: 3.0.0 (cocoapod)
 - Xcode Version: 8.3 (8E162)
 - Swift Version: Apple Swift version 3.1 (swiftlang-802.0.48 clang-802.0.38)
Target: x86_64-apple-macosx10.9

Please also mention which package manager you used and its version. Delete the
other package managers in this list:

 - Cocoapods: 1.1.1

### Project that demonstrates the issue

The line of code in SwiftJSON.swift is the setter for NSURL:

```
set {
    self.object = newValue?.absoluteString
}
```

And the compile is suggesting something like:

```
set {
    self.object = newValue?.absoluteString ?? NSNull()
}
```
Please link to a project we can download that reproduces the issue. Feel free
to delete this section if it's not relevant to the issue (eg - feature request).

The project should be [short, self-contained, and correct example](http://sscce.org/).

**Comment by wongzigii** [comment]
Hey @chessboy, thanks for filling an issue here. It seems like you are using an old version SwiftyJSON (3.0.0), which is not compatible with Xcode 8 and Swift 3.1.

**Comment by justinctlam** [comment]
@wongzigii Still occurs with the latest version I believe. It's a new compiler warning introduced with Xcode 8.3.

**Comment by wongzigii** [comment]
@justinctlam 
Are you getting the same compiler warning as this issue? 
Are you using CocoaPods or Carthage ?
What's the version of your dependency manager?

**Comment by justinctlam** [comment]
@wongzigii My apologies, it seems like the warning was fixed in 3.1.2 and I made the wrong assumption of the version we were using.  Thank you for following up, I appreciate all the hard work you guys are doing.

**Comment by wongzigii** [comment]
Great to hear that! So I assume this issue can be closed now. Feel free to open another issue if you have any problems.
